src/actions/notifications/index.ts

```diff
@@ -1,41 +1,136 @@
+import { ThunkAction } from 'src/store'
+import { Notification } from 'src/lib/notifications'
 import {
-  CLEAR_NOTIFICATIONS,
-  REMOVE_NOTIFICATION,
+  SET_ACTIVE_NOTIFICATION,
   SCHEDULE_NOTIFICATION,
-} from '../../reducers/notifications'
-import { randomBytes } from 'crypto'
-import {
-  Notification,
-  NotificationMessage,
-  NotificationScope,
-  NotificationSeverity,
-  NotificationType,
-} from '../../reducers/notifications/types'
-
-export const removeNotification = (notification: Notification) => ({
-  type: REMOVE_NOTIFICATION,
-  value: notification,
-})
+  REMOVE_NOTIFICATION,
+  CLEAR_NOTIFICATIONS,
+} from 'src/reducers/notifications'
 
-export const scheduleNotification = (notification: Notification) => ({
-  type: SCHEDULE_NOTIFICATION,
-  value: notification,
-})
 
-export const clearAllNotifications = () => ({
-  type: CLEAR_NOTIFICATIONS,
+export const setActiveNotification = (
+  notification: Notification | null,
+  expiry?: number,
+) => ({
+  type: SET_ACTIVE_NOTIFICATION,
+  notification,
+  expiry,
 })
 
-export const infoNotification = (
-  message: NotificationMessage,
-): Notification => ({
-  uid: randomBytes(4).toString('hex'), // TODO abstract
-  type: NotificationType.info,
-  message,
-  severity: NotificationSeverity.medium,
-  scope: NotificationScope.global,
-  dismissible: true,
-  autoDismissMs: 3000,
-  handleConfirm: removeNotification,
-  handleDismiss: removeNotification,
-})
+export const scheduleNotification = (
+  notification: Notification,
+): ThunkAction => dispatch => {
+  dispatch({
+    type: SCHEDULE_NOTIFICATION,
+    notification,
+  })
+  return dispatch(updateNotificationsState)
+}
+
+export const removeNotification = (
+  notification: Notification,
+): ThunkAction => dispatch => {
+  dispatch({
+    type: REMOVE_NOTIFICATION,
+    notification,
+  })
+  return dispatch(updateNotificationsState)
+}
+
+export const invokeInteract = (
+  notif: Notification,
+): ThunkAction => async dispatch => {
+  let keepNotification = false
+  const { interact } = notif
```

Comment:
Does `invokeInteract` and `invokeDismiss` need the `notification` as an argument if the notification will always be the `active` one from the store? Why not use `getState()`?

Reply:
it's "cleaner" this way, and was basically done like so to be more flexible (container can decide which/when/how notification actions are invoked), and also to avoid potentially weird race conditions (no tests to prove it, but I just feel safer like this....)